openpype/hosts/maya/plugins/publish/validate_rendersettings.py

```diff
@@ -305,6 +325,10 @@ def repair(cls, instance):
             "{aov_separator}", instance.data.get("aovSeparator", "_")
         )
 
+        for attribute, data in cls.get_nodes(instance, renderer).items():
+            for node in data["nodes"]:
+                lib.set_attribute(attribute, data["values"][0], node)
```

Comment:
I am not quite sure about this, does the setting only allow the attributes with string type? If not, the function of **set_attributes** needs some rewrites so that the render attributes with varied type can be repaired after the validation. RIght now it only allows the attributes with string type to be repaired. Otherwises, if I add something like `aiOptions.AA_samples`, it will give the RuntimeError and it can't perform the repairment as it is an attribute with integer type.
`
File "D:\toke\pype\build\exe.win-amd64-3.9\openpype\hosts\maya\api\lib.py", line 1395, in set_attribute
cmds.setAttr(node_attr, value, type=attr_type)
RuntimeError: setAttr: The attribute 'defaultArnoldRenderOptions.AASamples' does not accept data of type 'string'.
`

Reply:
Actually, since you brought this up. Some time ago this got reported #3890 as an issue and I [wrote this comment](https://github.com/ynput/OpenPype/issues/3890#issuecomment-1255167358) with some examples on how it can tackle different cases but actually ALSO do the validation correctly with e.g. booleans, etc. and it will allow a tolerance for float values since 1.0 in settings might be 1.000000000000123 when doing `getAttr` on it in Maya. Since booleans are basically supported due to the string conversion I made a commit that seemed to work [here](https://github.com/BigRoy/OpenPype/commit/5cd0069f1500bb278fbb6e40d8502cf41d035da2) as [mentioned in a later comment](https://github.com/ynput/OpenPype/issues/3890#issuecomment-1255578184)

Would this be the moment to consider those additional types like enum attributes, booleans, float value precisions, etc in our tests?

Reply:
> Would this be the moment to consider those additional types like enum attributes, booleans, float value precisions, etc in our tests?

Sorry, I think that is beyond the scope of this PR.